swarm/network/stream/snapshot_sync_test.go

```diff
@@ -0,0 +1,762 @@
+			}
+			if pl == 256 || pl == po {
+				log.Trace(fmt.Sprintf("appending %s", conf.addrToIdMap[string(a)]))
+				nns = append(nns, indexmap[string(a)])
+				nodemap[string(a)] = append(nodemap[string(a)], i)
+			}
+			if pl == 256 && len(nns) >= testMinProxBinSize {
+				//maxProxBinSize has been reached at this po, so save it
+				//we will add all other nodes at the same po
+				pl = po
+			}
+			return true
+		})
+		kmap[string(conf.chunks[i])] = nns
+	}
+	if log.Lvl(*loglevel) == log.LvlTrace {
+		for k, v := range nodemap {
+			log.Trace(fmt.Sprintf("Node %s: ", conf.addrToIdMap[k]))
+			for _, vv := range v {
+				log.Trace(fmt.Sprintf("%v", conf.chunks[vv]))
+			}
+			log.Trace(fmt.Sprintf("%v", conf.addrToIdMap[k]))
+		}
+		for k, v := range kmap {
+			log.Trace(fmt.Sprintf("Chunk %s: ", k))
+			for _, vv := range v {
+				log.Trace(fmt.Sprintf("%v", conf.addrToIdMap[string(conf.addrs[vv])]))
+			}
+		}
+	}
+	for addr, chunks := range nodemap {
+		//this selects which chunks are expected to be found with the given node
+		conf.idToChunksMap[conf.addrToIdMap[addr]] = chunks
+	}
+	log.Debug(fmt.Sprintf("Map of expected chunks by ID: %v", conf.idToChunksMap))
+	conf.chunksToNodesMap = kmap
+}
+
+//upload a file(chunks) to a single local node store
+func uploadFileToSingleNodeStore(id discover.NodeID, chunkCount int) ([]storage.Key, error) {
+	log.Debug(fmt.Sprintf("Uploading to node id: %s", id))
+	lstore := stores[id]
+	size := chunkSize
+	dpa := storage.NewDPA(lstore, storage.NewChunkerParams())
+	dpa.Start()
+	var rootkeys []storage.Key
+	for i := 0; i < chunkCount; i++ {
+		rk, wait, err := dpa.Store(io.LimitReader(crand.Reader, int64(size)), int64(size))
+		wait()
+		if err != nil {
+			return nil, err
+		}
+		rootkeys = append(rootkeys, (rk))
+	}
+
+	defer dpa.Stop()
+
+	return rootkeys, nil
+}
+
+//Here we wait until all connections from the snapshot are up
+func waitForSnapshotConnsUp(ctx context.Context, net *simulations.Network, done chan struct{}, connCount int, errc chan error) {
+	arrivedConns := 0
+	events := make(chan *simulations.Event)
+	//subscribe to all events from the network
+	sub := net.Events().Subscribe(events)
+	defer sub.Unsubscribe()
+
+	for {
+		select {
+		case <-ctx.Done():
+			errc <- fmt.Errorf("Timeout waiting for Snapshot connections")
+		case event := <-events:
+			//if the event is of type connection, is a Live event and the connection is up
+			//NOTE; this will require that all connections are UP in the snapshot!
+			if event.Type == simulations.EventTypeConn && !event.Control && event.Conn.Up {
+				arrivedConns++
+				//the amount of expected connections has been reached, so we can stop waiting
+				if arrivedConns == connCount {
+					done <- struct{}{}
+					return
+				}
+			}
+		case err := <-sub.Err():
+			if err != nil {
+				errc <- err
+			}
+		}
+	}
+}
+
+//initialize a network from a snapshot
+func initNetWithSnapshot(nodeCount int) (*simulations.Network, error) {
+
+	adapter := "sim"
```

Comment:
please take this from a flag, also loglevel

Reply:
This is actually already done in `common_test.go`, so going to use that code